### PR TITLE
add nextAuth

### PR DESCRIPTION
**Pull request by Roland-Hufnagel**

### CODE REVIEW

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Looking already very good, I like how you use early returns, I found some places where you can implement it as well.

File: components/Header.js [hunk 1]
```diff
@@ -1,39 +1,45 @@
...
+          src="/img/mogen.png"
+          alt="lots of feedback"
+          layout="fixed"
+          width={65}
+          height={65}
+        />
+      </span>
+      <span>
+        <LoginButton />
+      </span>
```
[the remark this answers is not in the document]
Comment (Roland-Hufnagel) [reply components/Header.js:33]: The styling was not very good at all. I changed to a Login-Area now.

File: pages/_app.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
 import GlobalStyle from "../components/GlobalStyle";
 import Layout from "../components/Layout";
+import { SessionProvider } from "next-auth/react";
 
-function MyApp({ Component, pageProps }) {
+function MyApp({ Component, pageProps: { session, ...pageProps } }) {
```
[the remark this answers is not in the document]
Comment (Roland-Hufnagel) [reply pages/_app.js:5]: I tried to change but I found out: 
It is the Rest-Operator here and so it has to be at the end. This is the way as described in the next-auth docs.

File: pages/api/create/index.js [hunk 3]
```diff
@@ -5,22 +5,23 @@ import { authOptions } from "../auth/[...nextauth]";
...
+    postData.owner = session.user.email;
+    const newSurvey = await SurveyModel.create(postData);
+    await SurveyModel.findByIdAndUpdate(newSurvey.id, {
+      url: newSurvey.url + newSurvey.id,
+    });
+    return res
+      .status(201)
+      .json({ message: "Survey created", createdId: newSurvey.id });
+  } else {
+    return res.status(405).json({ message: "HTTP Method not allowed" });
```
Comment (F-Kirchhoff) [inline pages/api/create/index.js:25]: please don't return a statement that is not handled outside of the function:
```suggestion
    res.status(405).json({ message: "HTTP Method not allowed" });
    return 
```

File: pages/api/edit/[id].js [hunk 4]
```diff
@@ -5,22 +5,25 @@ import { authOptions } from "../auth/[...nextauth]";
...
+      .status(200)
+      .json({ message: "Survey deleted", deletedId: survey.id });
+  } else if (req.method === "PATCH") {
+    const postData = JSON.parse(request.body);
+    const newSurvey = await SurveyModel.findByIdAndUpdate(id, postData);
+    return res
+      .status(200)
+      .json({ message: "Survey updated", updatedId: newSurvey.id });
+  } else {
+    return res.status(405).json({ message: "HTTP method not allowed" });
```
Comment (F-Kirchhoff) [inline pages/api/edit/[id].js:27]: here as welll

File: pages/api/edit/[id].js [hunk 5]
```diff
@@ -5,22 +5,25 @@ import { authOptions } from "../auth/[...nextauth]";
...
+  await dbConnect();
+  if (req.method === "DELETE") {
+    const survey = await SurveyModel.findByIdAndDelete(id);
+    return res
+      .status(200)
+      .json({ message: "Survey deleted", deletedId: survey.id });
+  } else if (req.method === "PATCH") {
+    const postData = JSON.parse(request.body);
+    const newSurvey = await SurveyModel.findByIdAndUpdate(id, postData);
+    return res
```
Comment (F-Kirchhoff) [inline pages/api/edit/[id].js:23]: here as well

File: pages/api/create/index.js [hunk 6]
```diff
@@ -5,22 +5,23 @@ import { authOptions } from "../auth/[...nextauth]";
...
+
+  if (req.method === "POST") {
+    await dbConnect();
+    const postData = JSON.parse(req.body);
+    postData.owner = session.user.email;
+    const newSurvey = await SurveyModel.create(postData);
+    await SurveyModel.findByIdAndUpdate(newSurvey.id, {
+      url: newSurvey.url + newSurvey.id,
+    });
+    return res
```
Comment (F-Kirchhoff) [inline pages/api/create/index.js:21]: here as well

**Review by F-Kirchhoff — COMMENTED** [review]
LGTM 🦊

**Pull request merged**